### DISCUSSION TITLE
关于indexFile和consumeQueue的数据完整性问题

### DISCUSSION THREAD

**Discussion by ijbuhv** [apache/rocketmq#7123] [Q&A]
commitLog用crc来验证了数据的完整,并且在重启恢复时抛弃了不完整的数据,但是indexFile和consumeQueue的数据没有看到有数据完整性验证,那么是怎么保证数据不出错的呢

**Comment by HScarb** [comment]
IndexFile和ConsumeQueue刷盘时都会将`StoreCheckpoint`刷盘，它用来记录当前这两个文件的刷盘进度。
重启时从上次保存的刷盘进度开始，通过CommitLog重新构建。
只要保证CommitLog完整，就可以构建出一模一样的IndexFile和ConsumeQueue

**Comment by ijbuhv** [comment]
![image](https://github.com/apache/rocketmq/assets/141534357/dfd12d4c-3c3f-492e-b8f3-1031dd164f8a)

我看indexFile会拒绝超过IndexHeader记录的physicalOffset的写入,会不会有indexFile的indexHeader刷到磁盘,但是slot数据没有刷到磁盘,这个时候利用commitLog重建会不会重建有丢失呢
